transports/quic/src/transport.rs

```diff
@@ -0,0 +1,641 @@
+// Copyright 2017-2020 Parity Technologies (UK) Ltd.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+//! Implementation of the [`Transport`] trait for QUIC.
+//!
+//! Combines all the objects in the other modules to implement the trait.
+
+use crate::connection::Connection;
+use crate::endpoint::ToEndpoint;
+use crate::{endpoint::EndpointChannel, muxer::QuicMuxer, upgrade::Upgrade};
+use crate::{Config, ConnectionError};
+
+#[cfg(feature = "async-std")]
+mod async_std;
+#[cfg(feature = "tokio")]
+mod tokio;
+#[cfg(feature = "async-std")]
+pub use async_std::{AsyncStd, AsyncStdTransport};
+use futures::future::BoxFuture;
+#[cfg(feature = "tokio")]
+pub use tokio::{Tokio, TokioTransport};
+
+use futures::channel::{mpsc, oneshot};
+use futures::ready;
+use futures::stream::StreamExt;
+use futures::{prelude::*, stream::SelectAll};
+
+use if_watch::{IfEvent, IfWatcher};
+
+use libp2p_core::{
+    multiaddr::{Multiaddr, Protocol},
+    transport::{ListenerId, TransportError as CoreTransportError, TransportEvent},
+    PeerId, Transport,
+};
+use std::collections::VecDeque;
+use std::io;
+use std::marker::PhantomData;
+use std::net::{IpAddr, UdpSocket};
+use std::task::Waker;
+use std::{
+    net::SocketAddr,
+    pin::Pin,
+    task::{Context, Poll},
+};
```

Comment:
I'd prefer if we merge all `use` statements into one block.

Reply:
Not strong opinion on my side either. I usually group imports myself as well, but this was existing code so I did not bother to change it. I wonder if this is really something that we have to be strict about?

Reply:
I am not too fussed. It is mostly the module declarations half-way through that confuse me :)

Reply:
You're right, they should not be hidden in between import statements! Will fix it.